components/kyma-environment-broker/internal/broker/logging.go

```diff
@@ -0,0 +1,48 @@
+package broker
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+var openKeys = map[string]struct{}{
+	"sm_url":           {},
+	"xsappname":        {},
+	"globalaccount_id": {},
+	"subaccount_id":    {},
+}
+
+func hideSensitiveDataFromRawContext(d []byte) map[string]interface{} {
+	var data map[string]interface{}
+	err := json.Unmarshal(d, &data)
+	if err != nil {
+		return map[string]interface{}{}
+	}
+	return hideSensitiveDataFromContext(data)
+}
+
+func hideSensitiveDataFromContext(input map[string]interface{}) map[string]interface{} {
+	copy := input
```

Comment:
nit: this actually doesn't copy the map, only creates a new reference for the same memory where the map is stored. Also, maps are passed to functions by reference, so ` hideSensitiveDataFromContext()` will modify the input values in place. But given it operates on freshly unmarshalled structs not re-used in the rest of the code, it doesn't cause problems.